File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for (int i = 0; i < priorities.length; i++) {
            pq.add(priorities[i]);
        }
        int cnt = 0;
        while (!pq.isEmpty()) {
            for (int i = 0; i < priorities.length; i++) {
                if (priorities[i] == pq.peek()) {
                    cnt += 1;
                    pq.poll();
                    if (location == i) {
                        return cnt;
                    }
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.kt
import java.util.*

class Solution {
    fun solution(priorities: IntArray, location: Int): Int {
        var answer = 0
        val q = LinkedList<Pair<Int, Int>>()
        var idx = 0
        for (priority in priorities) {
            q.add(Pair(idx, priority))
            idx += 1
        }
        while(q.isNotEmpty()) {
            val current = q.remove()
            val (index, priority) = current
            
            if (q.any{it.second > priority}) {
                q.add(current)
            } else {
                answer += 1
                if (index == location) {
                    return answer
                }
            }
        }
        return answer
    }
}